src/deployment/registration.py

```diff
@@ -221,7 +224,9 @@ def create_application_registration(
     return registered_app
 
 
-def add_application_password(app_object_id: UUID) -> Tuple[str, str]:
+def add_application_password(
+    app_object_id: UUID, subscription_id: str
```

Comment:
Can the `subscription_id` be narrowed to a `UUID`, too?

Reply:
Not without covering the code with casts back and forth.  All of the underlying Azure APIs expect it as a string.